sql/catalog: add a descriptor for session-based system.leases table

fqazi opened the pull request:
This patch adds a new descriptor that can be used to read/write to a session-based leasing table. It also has a table crdb_internal.kv_session_based_leases for reading from the modified lease table format. This format is also added to the debug zip but will be populated later. Additionally, the second commit in this PR updates the lease KVWriter and its tests for writing to the new table format, but these changes make no functional changes to the actual leasing mechanism.

informs: https://github.com/cockroachdb/cockroach/issues/95765
Epic: CC-24173
Release note: None


The first two commits can be ignored and belong to the following PRs: 
https://github.com/cockroachdb/cockroach/pull/111345
https://github.com/cockroachdb/cockroach/pull/111427

fqazi committed:
sql/catalog: add a descriptor for session-based system.leases table

This patch adds a new descriptor that can be used to
read/write to a session-based leasing table. It also
has a table crdb_internal.kv_session_based_leases for
reading from the modified lease table format. This format
is also added to the debug zip but will not be populated yet.

Epic: CC-24173

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/111462)
<!-- Reviewable:end -->

fqazi committed:
catalog/lease: add kvwriter support for session based leases

This commit teaches the `KVWriter` how to write to the incipient session
based leasing format.  The ability to do so is gated by the session
based leasing mode cluster setting. Therefore this change is only
utilized in unit test cases.

Epic: CC-24173

Release note: None

abarganier reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DrewKimball and @fqazi)

___
*[`pkg/cli/zip_table_registry.go` line 656 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-Nfl_kLA3TlyuPfHLvXf:-Nfl_kLB4gArzvFdGLbs:bkk5i3c) ([raw file](https://github.com/cockroachdb/cockroach/blob/073b143b79406da9ddee1466a382352d85b3ffb5/pkg/cli/zip_table_registry.go#L656)):*
> ```Go
> 	"crdb_internal.kv_session_based_leases": {
> 		nonSensitiveCols: NonSensitiveColumns{
> 			"\"descID\"",
> ```

Are the escaped quotes here necessary? This would be the firs table in the registry that does this, AFAIK, so wanted to check. 


<!-- Sent from Reviewable.io -->

fqazi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier and @DrewKimball)

___
*[`pkg/cli/zip_table_registry.go` line 656 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-Nfl_kLA3TlyuPfHLvXf:-NfpwCB0EK_H2sxllOET:b-ob0fnu) ([raw file](https://github.com/cockroachdb/cockroach/blob/073b143b79406da9ddee1466a382352d85b3ffb5/pkg/cli/zip_table_registry.go#L656)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

Are the escaped quotes here necessary? This would be the firs table in the registry that does this, AFAIK, so wanted to check.
</blockquote></details>

system.lease has the same thing, it's because the columns are weirdly uppercase. However, I can remove the escaping and use ` instead like that table.


<!-- Sent from Reviewable.io -->

abarganier reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DrewKimball and @fqazi)

___
*[`pkg/cli/zip_table_registry.go` line 656 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-Nfl_kLA3TlyuPfHLvXf:-NgKW2Ek5LqnGPKpcm4p:b-sfpf97) ([raw file](https://github.com/cockroachdb/cockroach/blob/073b143b79406da9ddee1466a382352d85b3ffb5/pkg/cli/zip_table_registry.go#L656)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

system.lease has the same thing, it's because the columns are weirdly uppercase. However, I can remove the escaping and use ` instead like that table.
</blockquote></details>

Ack, sounds good, was mostly just curious. Thanks for elaborating!


<!-- Sent from Reviewable.io -->

chrisseto commented on pkg/sql/catalog/lease/kv_writer.go:
```diff
@@ -66,7 +74,17 @@ func (w *kvWriter) versionGuard(
 func (w *kvWriter) insertLease(ctx context.Context, txn *kv.Txn, l leaseFields) error {
 	return w.do(ctx, txn, l, func(guard settingswatcher.VersionGuard, b *kv.Batch) error {
-		if guard.IsActive(clusterversion.TODO_Delete_V23_1_SystemRbrDualWrite) {
+		// Write a session based lease, if at the very least we are dual writing
```
I'm finding this comment (and the similar one below) very difficult to interpret. I generally expect a comment ahead of an if to be in the format `why, action` as that matches the flow of control. Alternatively, `action, why` can be placed within the if block.

The `isSessionBasedLeasingModeActive` method is also quite confusing as it implies that the mode is an enum but the comment states `at the very least`. You could beef up the comment here to clear up any confusion but I think a bit of renaming or refactoring could go a long way.

This is probably too much of a refactor at this point but a bitset feels like a great tool for this type of job:

```go
ReadExpiration = 0b0001
WriteExpiration = 0b0010
ReadSession = 0b0100
WriteSession = 0b0100

DualWrite = WriteExpiration | WriteSession

if mode & WriteExpiration == WriteExpiration {
...
}
```

Maybe just rename `isSessionBasedLeasingModeActive` to `sessionBasedLeasingModeIsAtLeast`?

chrisseto reviewed (commented):
Can't easily comment on this in github but the grammar in the message of c912124dbc19867ab52884fabfe6d8e2dedb156f is a bit dicey.

How about:
```
This commit teaches the `KVWriter` how to write to the incipient session based leasing format.
The ability to do so is gated by the session based leasing mode cluster setting. Therefore this change is only utilized in unit test cases.
```

Other than that my only blocking comment is around the span lines disappearing.

fqazi committed:
catalog/lease: rename isSessionBasedLeasingModeActive

Prviously, the method isSessionBasedLeasingModeActive was not clear in
describing which leasing mode is active. This patch renames
isSessionBasedLeasingModeActive ta sessionBasedLeasingModeIsAtLeast

Release note: None

fqazi reviewed (commented):
Updated the commit message and appreciate the review Chris :)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-:-NjSqCPq4DIlK-cypn7r:b-hwb4x1)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @chrisseto, and @DrewKimball)

___
*[`pkg/cli/zip_table_registry.go` line 656 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-Nfl_kLA3TlyuPfHLvXf:-NjSqEOoBAQXbzGKInh_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/073b143b79406da9ddee1466a382352d85b3ffb5/pkg/cli/zip_table_registry.go#L656)):*
<details><summary><i>Previously, abarganier (Alex Barganier) wrote…</i></summary><blockquote>

Ack, sounds good, was mostly just curious. Thanks for elaborating!
</blockquote></details>

Done.
___
*[`pkg/sql/catalog/lease/kv_writer.go` line 33 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387070617:-NjSqFPv2s3KCSypZvHl:b-1vp8pt) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer.go#L33)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

Could we toss in some comments explaining what the various writers here are? New and old are _kinda_ self explanatory but we now have 3 potential options?
</blockquote></details>

Done, and the oldWriter is also dead now its called expiryBased and sessionBased to avoid confusion.
___
*[`pkg/sql/catalog/lease/kv_writer.go` line 77 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387087434:-NjT2MvKEO1ldL9Xz6Nh:bhgpenz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer.go#L77)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

I'm finding this comment (and the similar one below) very difficult to interpret. I generally expect a comment ahead of an if to be in the format `why, action` as that matches the flow of control. Alternatively, `action, why` can be placed within the if block.

The `isSessionBasedLeasingModeActive` method is also quite confusing as it implies that the mode is an enum but the comment states `at the very least`. You could beef up the comment here to clear up any confusion but I think a bit of renaming or refactoring could go a long way.

This is probably too much of a refactor at this point but a bitset feels like a great tool for this type of job:

```go
ReadExpiration = 0b0001
WriteExpiration = 0b0010
ReadSession = 0b0100
WriteSession = 0b0100

DualWrite = WriteExpiration | WriteSession

if mode & WriteExpiration == WriteExpiration {
...
}
```

Maybe just rename `isSessionBasedLeasingModeActive` to `sessionBasedLeasingModeIsAtLeast`?
</blockquote></details>

// We support writing both session based and expiry based leases within  
// the KV writer. To be able to support a migration between the two types  
// of writer will in some cases need to be able to write both types of leases.  
// As a result based on our currently active mode, determine which types  
// of leases should be written. The scenarios we support are:  
// 1) Session Based Off => Only expiry based leases are written.  
// 2) Dual-Write => Both session and expiry based leases will be written.  
// 3) Session Only => Only session based leases will get written.
___
*[`pkg/sql/logictest/testdata/logic_test/pg_builtins` line 0 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387063785:-NjSrGmqCP_fGOUhxv9P:b-tme3ws) ([raw file](https://github.com/cockroachdb/cockroach/blob/f7ea54911911091283eb952cbe03ee30e74e352a/pkg/sql/logictest/testdata/logic_test/pg_builtins#L0)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/cockroachdb/cockroach/pull/111462#discussion_r1387063785) to the right spot — sorry!)*

Is it normal to see all these oids change? That seems undesirable but the alphabetization of `constants.go` suggests it's expected.
</blockquote></details>

Yeah this is expected
___
*[`pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant` line 0 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387062472:-NjSu51zDFCPl8aGO29t:b-3p322o) ([raw file](https://github.com/cockroachdb/cockroach/blob/f7ea54911911091283eb952cbe03ee30e74e352a/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant#L0)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/cockroachdb/cockroach/pull/111462#discussion_r1387062472) to the right spot — sorry!)*

Is this diff expected? It seems unrelated to the changes made in this commit.
</blockquote></details>

Done.

No, and I cleaned it up. Not sure why my rewrite hit this one before.
___
*[`pkg/sql/catalog/lease/kv_writer_test.go` line 62 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387045695:-NjSvmoq49XNUaM1DIn6:bd8oe3x) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer_test.go#L62)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

Do we want to run this test in the dual mode as well?
</blockquote></details>

I think for simplicity it's easier to test them independently, since the state is tied to the descriptors. As we enable it in the leasing subsystem we'll start covering the mixed mode where need to use both descriptor versions to count.
___
*[`pkg/sql/catalog/lease/lease_test.go` line 3471 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387065917:-NjSt3jm2_yTUE9OALF8:b-6thrzr) ([raw file](https://github.com/cockroachdb/cockroach/blob/f7ea54911911091283eb952cbe03ee30e74e352a/pkg/sql/catalog/lease/lease_test.go#L3471)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

Would it be helpful/useful to add in a row in the old format and verify that that can be read as well? I'm expecting that session ID would just be null but is that the case?
</blockquote></details>

Not really, because the new descriptor is only for reading / writing rows in the new format. i.e. The old descriptor is used through out the code / unit tests so it has some coverage.
___
*[`pkg/sql/catalog/lease/lease_test.go` line 3460 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387049840:-NjSswnd59VVRxiGIjvW:b-j1o44b) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/lease_test.go#L3460)):*
<details><summary><i>Previously, chrisseto (Chris Seto) wrote…</i></summary><blockquote>

For my own education, what is the purpose of this test and internal function? Is it just a way to validate that migrations work as expected ahead of actually applying them?
</blockquote></details>

Yeah, this is a sanity check that the descriptor format is sane and we can read / write rows with it. We haven't actually changed the system descriptor definition yet.


<!-- Sent from Reviewable.io -->

fqazi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abarganier, @chrisseto, and @DrewKimball)

___
*[`pkg/sql/catalog/lease/kv_writer.go` line 77 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387087434:-NjT2WJBDIjqNhlGdY7Z:b260pd3) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer.go#L77)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

// We support writing both session based and expiry based leases within  
// the KV writer. To be able to support a migration between the two types  
// of writer will in some cases need to be able to write both types of leases.  
// As a result based on our currently active mode, determine which types  
// of leases should be written. The scenarios we support are:  
// 1) Session Based Off => Only expiry based leases are written.  
// 2) Dual-Write => Both session and expiry based leases will be written.  
// 3) Session Only => Only session based leases will get written.
</blockquote></details>

Renamed this and change the comment to the one above to make it clear what the pattern is


<!-- Sent from Reviewable.io -->

chrisseto reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

The comment updates are incredible!

![Pacha Just Right GIF - Pacha Just Right So Good - Discover & Share GIFs](https://media.tenor.com/4dz1iXzgfpAAAAAC/pacha-just-right.gif)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-:-Njnjr2KC_9vto8mKn8_:b-f0d7sp)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier and @DrewKimball)

___
*[`pkg/sql/catalog/lease/kv_writer.go` line 33 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387070617:-NjnjDsU3Q79VRz0Eydo:b-romc63) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer.go#L33)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Done, and the oldWriter is also dead now its called expiryBased and sessionBased to avoid confusion.
</blockquote></details>

Oh nice, thanks!
___
*[`pkg/sql/catalog/lease/kv_writer.go` line 77 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387087434:-NjnjPAT0Kz_UdUQNavg:b4putsz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer.go#L77)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Renamed this and change the comment to the one above to make it clear what the pattern is
</blockquote></details>

:ok_hand: 
___
*[`pkg/sql/catalog/lease/kv_writer_test.go` line 62 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387045695:-NjnjdcODpJC_SysyQrg:bb6sh25) ([raw file](https://github.com/cockroachdb/cockroach/blob/c912124dbc19867ab52884fabfe6d8e2dedb156f/pkg/sql/catalog/lease/kv_writer_test.go#L62)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

I think for simplicity it's easier to test them independently, since the state is tied to the descriptors. As we enable it in the leasing subsystem we'll start covering the mixed mode where need to use both descriptor versions to count.
</blockquote></details>

Ack!
___
*[`pkg/sql/catalog/lease/lease_test.go` line 3471 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/111462#gh-1387065917:-NjnjkvLEYo6IRzDm5ns:b-4gor3o) ([raw file](https://github.com/cockroachdb/cockroach/blob/f7ea54911911091283eb952cbe03ee30e74e352a/pkg/sql/catalog/lease/lease_test.go#L3471)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Not really, because the new descriptor is only for reading / writing rows in the new format. i.e. The old descriptor is used through out the code / unit tests so it has some coverage.
</blockquote></details>

Gotcha!


<!-- Sent from Reviewable.io -->

fqazi commented:
@chrisseto Thanks for the review ☺️

bors r+

rafiss reviewed (commented):
> Reviewed 9 of 42 files at r6, 17 of 27 files at r8, 2 of 9 files at r11, 7 of 7 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[`pkg/sql/crdb_internal.go` line 1559 at r11](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtH5hU0tRzX83tLLE5:-NjtH5hU0tRzX83tLLE6:b-kn3kzd) ([raw file](https://github.com/cockroachdb/cockroach/blob/d040e0f6cff23bfd45c733551a75e5175377b5ce/pkg/sql/crdb_internal.go#L1559)):*
> ```Go
> 	schema: `
> CREATE TABLE crdb_internal.kv_session_based_leases (
>   "descID"     INT8,
> ```

we should avoid using the mixed case names whenever possible. they add a bit of friction for anyone wanting to use this.
___
*[`pkg/sql/crdb_internal.go` line 1570 at r11](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtH88uCSV3lp31ehSx:-NjtH88uCSV3lp31ehSy:b-8rfdk0) ([raw file](https://github.com/cockroachdb/cockroach/blob/d040e0f6cff23bfd45c733551a75e5175377b5ce/pkg/sql/crdb_internal.go#L1570)):*
> ```Go
> 		return p.InternalSQLTxn().WithSyntheticDescriptors(catalog.Descriptors{systemschema.LeaseTable_V24_1()},
> 			func() error {
> 				rows, err := p.InternalSQLTxn().QueryBuffered(ctx, "query-leases", p.Txn(), "SELECT * FROM system.lease")
> ```

we should never use `SELECT *` in internal queries. they can break easily if the table changes.
___
*[`pkg/sql/catalog/lease/kv_writer_test.go` line 63 at r13](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtIHBS7tKlGRD926bv:-NjtIHBS7tKlGRD926bw:b-klwte1) ([raw file](https://github.com/cockroachdb/cockroach/blob/66567b7dfd4722844a54d71360cf2bfaa6c1bb44/pkg/sql/catalog/lease/kv_writer_test.go#L63)):*
> ```Go
> 
> 	for _, mode := range []SessionBasedLeasingMode{SessionBasedLeasingOff, SessionBasedOnly} {
> 		t.Run(fmt.Sprintf("Session based leasing mode: %v", mode), func(t *testing.T) {
> ```

nit: this makes the test name a bit verbose
___
*[`pkg/sql/catalog/lease/kv_writer_test.go` line 233 at r13](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtIQXeEZYKRCJ16GzV:-NjtIQXeEZYKRCJ16GzW:b-rwcp0d) ([raw file](https://github.com/cockroachdb/cockroach/blob/66567b7dfd4722844a54d71360cf2bfaa6c1bb44/pkg/sql/catalog/lease/kv_writer_test.go#L233)):*
> ```Go
> 			instanceID:   base.SQLInstanceID(rand.Intn(vals)),
> 			expiration:   *ts,
> 			sessionID:    []byte(ts.String()),
> ```

why are we using the timestamp as a session ID here?


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[`pkg/sql/crdb_internal.go` line 1570 at r11](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtH88uCSV3lp31ehSx:-NjtJ2Dg1OnW30PrFszF:bwl6qn6) ([raw file](https://github.com/cockroachdb/cockroach/blob/d040e0f6cff23bfd45c733551a75e5175377b5ce/pkg/sql/crdb_internal.go#L1570)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

we should never use `SELECT *` in internal queries. they can break easily if the table changes.
</blockquote></details>

actually, these should be implemented as a virtual view rather than as a virtual table. i can make this change.


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[`pkg/sql/crdb_internal.go` line 1570 at r11](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtH88uCSV3lp31ehSx:-NjtJge5Dz5SdLjJBlTh:b-vrrnyo) ([raw file](https://github.com/cockroachdb/cockroach/blob/d040e0f6cff23bfd45c733551a75e5175377b5ce/pkg/sql/crdb_internal.go#L1570)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

actually, these should be implemented as a virtual view rather than as a virtual table. i can make this change.
</blockquote></details>

oh hm, maybe they can't since it needs the synthetic descriptors


<!-- Sent from Reviewable.io -->

fqazi commented:
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111462)_ status: :shipit: complete! 1 of 0 LGTMs obtained
> 
> _[`pkg/sql/crdb_internal.go` line 1570 at r11](https://reviewable.io/reviews/cockroachdb/cockroach/111462#-NjtH88uCSV3lp31ehSx:-NjtJge5Dz5SdLjJBlTh:b-vrrnyo) ([raw file](https://github.com/cockroachdb/cockroach/blob/d040e0f6cff23bfd45c733551a75e5175377b5ce/pkg/sql/crdb_internal.go#L1570)):_
> 
> Previously, rafiss (Rafi Shamim) wrote…
> oh hm, maybe they can't since it needs the synthetic descriptors

Let me open a PR to address those comments
